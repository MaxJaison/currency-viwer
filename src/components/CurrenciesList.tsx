import '../components/CurrencciesList.css';
import { CURRENCIES } from '../constants.ts';
import { CommonProps } from '../types.ts';

const CurrenciesList = ({ selectedCurrencies, selectCurrency }: CommonProps) => (
    <div className="currencies-list">
        {CURRENCIES.map((currency: string, index: number) => {
            return (
                <button key={index} className={selectedCurrencies.includes(currency) ? 'btn active' : 'btn'} onClick={() => selectCurrency(currency)}>
                    {currency}
                    <input className="checkbox" disabled type="checkbox" id={currency} checked={selectedCurrencies.includes(currency)} />
                    <label className="label" htmlFor={currency}></label>
                </button>
            );
        })}
    </div>
);

export default CurrenciesList;
