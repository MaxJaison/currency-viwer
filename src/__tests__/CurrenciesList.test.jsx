import { expect, test, vi } from 'vitest';
import { fireEvent, render } from '@testing-library/react';
import CurrenciesList from '../components/CurrenciesList.tsx';

const mockCallback = vi.fn();

test('check if currency button clicked', async () => {
    const currenciesList = render(<CurrenciesList selectedCurrencies={[]} selectCurrency={mockCallback} />);
    const currencyButton = await currenciesList.findByText('EUR');
    fireEvent.click(currencyButton);
    expect(mockCallback).toBeCalled();
    currenciesList.unmount();
});

test('create CurrenciesList snapshot', async () => {
    const { asFragment } = render(<CurrenciesList selectedCurrencies={[]} selectCurrency={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
});
