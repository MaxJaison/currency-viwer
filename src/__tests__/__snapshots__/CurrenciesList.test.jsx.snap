// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`create CurrenciesList snapshot 1`] = `
<DocumentFragment>
  <div
    class="currencies-list"
  >
    <button
      class="btn"
    >
      EUR
      <input
        class="checkbox"
        disabled=""
        id="EUR"
        type="checkbox"
      />
      <label
        class="label"
        for="EUR"
      />
    </button>
    <button
      class="btn"
    >
      PLN
      <input
        class="checkbox"
        disabled=""
        id="PLN"
        type="checkbox"
      />
      <label
        class="label"
        for="PLN"
      />
    </button>
    <button
      class="btn"
    >
      GEL
      <input
        class="checkbox"
        disabled=""
        id="GEL"
        type="checkbox"
      />
      <label
        class="label"
        for="GEL"
      />
    </button>
    <button
      class="btn"
    >
      DKK
      <input
        class="checkbox"
        disabled=""
        id="DKK"
        type="checkbox"
      />
      <label
        class="label"
        for="DKK"
      />
    </button>
    <button
      class="btn"
    >
      CZK
      <input
        class="checkbox"
        disabled=""
        id="CZK"
        type="checkbox"
      />
      <label
        class="label"
        for="CZK"
      />
    </button>
    <button
      class="btn"
    >
      GBP
      <input
        class="checkbox"
        disabled=""
        id="GBP"
        type="checkbox"
      />
      <label
        class="label"
        for="GBP"
      />
    </button>
    <button
      class="btn"
    >
      SEC
      <input
        class="checkbox"
        disabled=""
        id="SEC"
        type="checkbox"
      />
      <label
        class="label"
        for="SEC"
      />
    </button>
    <button
      class="btn"
    >
      USD
      <input
        class="checkbox"
        disabled=""
        id="USD"
        type="checkbox"
      />
      <label
        class="label"
        for="USD"
      />
    </button>
    <button
      class="btn"
    >
      RUB
      <input
        class="checkbox"
        disabled=""
        id="RUB"
        type="checkbox"
      />
      <label
        class="label"
        for="RUB"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`create a snapshot 1`] = `
<DocumentFragment>
  <div
    class="currencies-list"
  >
    <button
      class="btn"
    >
      EUR
      <input
        class="checkbox"
        disabled=""
        id="EUR"
        type="checkbox"
      />
      <label
        class="label"
        for="EUR"
      />
    </button>
    <button
      class="btn"
    >
      PLN
      <input
        class="checkbox"
        disabled=""
        id="PLN"
        type="checkbox"
      />
      <label
        class="label"
        for="PLN"
      />
    </button>
    <button
      class="btn"
    >
      GEL
      <input
        class="checkbox"
        disabled=""
        id="GEL"
        type="checkbox"
      />
      <label
        class="label"
        for="GEL"
      />
    </button>
    <button
      class="btn"
    >
      DKK
      <input
        class="checkbox"
        disabled=""
        id="DKK"
        type="checkbox"
      />
      <label
        class="label"
        for="DKK"
      />
    </button>
    <button
      class="btn"
    >
      CZK
      <input
        class="checkbox"
        disabled=""
        id="CZK"
        type="checkbox"
      />
      <label
        class="label"
        for="CZK"
      />
    </button>
    <button
      class="btn"
    >
      GBP
      <input
        class="checkbox"
        disabled=""
        id="GBP"
        type="checkbox"
      />
      <label
        class="label"
        for="GBP"
      />
    </button>
    <button
      class="btn"
    >
      SEC
      <input
        class="checkbox"
        disabled=""
        id="SEC"
        type="checkbox"
      />
      <label
        class="label"
        for="SEC"
      />
    </button>
    <button
      class="btn"
    >
      USD
      <input
        class="checkbox"
        disabled=""
        id="USD"
        type="checkbox"
      />
      <label
        class="label"
        for="USD"
      />
    </button>
    <button
      class="btn"
    >
      RUB
      <input
        class="checkbox"
        disabled=""
        id="RUB"
        type="checkbox"
      />
      <label
        class="label"
        for="RUB"
      />
    </button>
  </div>
</DocumentFragment>
`;
